### PR TITLE
Added flood prediction comparative ML system

### LINKED ISSUE #1437: Flood prediction system ,a comparative ML Models analysis and classification system
## Project Title
Flood Prediction System 

## Problem Statement
Floods are among the most common and devastating natural disasters, causing severe damage to human life, agriculture, infrastructure, and the environment.
Early flood prediction systems can help improve disaster preparedness, environmental monitoring, and decision-making processes.

## What This Project Does
This project focuses on predicting flood occurrence using environmental and rainfall-related  data-parameters through multiple Machine Learning algorithms.

The project includes:
- Data preprocessing and cleaning
- Exploratory Data Analysis (EDA)
- Feature scaling and transformation
- Training and comparison of multiple ML models including:
  - Logistic Regression
  - Support Vector Machine (SVM)
  - XGBoost
- Model evaluation using:
  - Accuracy
  - Precision
  - Recall
  - F1-score
- Data visualization and comparative performance analysis
- Saved trained models using Joblib
- Beginner-friendly modular implementation and documentation

## Dataset Information
The dataset contains environmental and rainfall-related attributes used for flood prediction and classification.
The dataset will be used for:
- Data cleaning and preprocessing
- Feature analysis
- Model training and testing
- Performance evaluation and comparison

## Real-World Applications
- Disaster preparedness and management
- Environmental monitoring systems
- Rainfall and flood risk assessment
- Agricultural and infrastructure planning
- Data-driven environmental analysis

## Tech Stack
Python, Pandas, NumPy, Scikit-learn, XGBoost, Matplotlib, Seaborn, Joblib

This project helps contributors understand and learn Machine Learning classification and regression workflows using a predefined environmental dataset.
I am a GSSoC 2026 contributor and would like to work on this project. Please assign this issue to me under GSSoC 2026.

### PR DESCRIPTION
**Pull request by v4rshh**
## Description

This Pull Request updates and refactors the **Flood Prediction System** to support dynamic web-based dataset loading via Kagglehub, modular machine learning pipelines, clean Git tracking, and a unified Master Jupyter Notebook.

### Features Implemented

* Flood prediction using environmental and rainfall-related parameters
* Comparative analysis of multiple ML classification algorithms:
  * Logistic Regression
  * Support Vector Machine (SVM)
  * Random Forest Classifier
  * XGBoost Classifier
  
* Model evaluation using:
  * Accuracy
  * F1-Score
  * Confusion Matrix
   
### Benefits of the Project

* Demonstrates practical implementation of multiple ML classification algorithms in one project
* Helps beginners understand comparative model analysis and evaluation techniques
* Provides visual insights into model performance and feature importance
* Can serve as a foundational framework for disaster prediction and environmental monitoring systems
* Encourages experimentation with preprocessing, feature engineering, and model optimization
* Offers reusable modular scripts for training and prediction workflows

### Visual Outputs Added

* Confusion Matrix plots
* Feature Importance graphs
* Comparative model performance analysis

### 1.  Dynamic Dataset Ingestion (`kagglehub`)
- Replaced local `.csv` dependency with dynamic fetching via Kaggle's official `kagglehub` library.
- Datasets are automatically fetched and cached on the fly:
  - **Repo 1**: [`varshithnagubandi/flood-prediction`](https://www.kaggle.com/datasets/varshithnagubandi/flood-prediction)
  - **Repo 2**: [`varshithnagubandi/indian-flood-prediction`](https://www.kaggle.com/datasets/varshithnagubandi/indian-flood-prediction)
- Added `.gitignore` configurations to exclude raw dataset CSVs and heavy `.joblib` binary artifacts from Git tracking.

### 2.  Complete Master Notebook Architecture
- Created a single, self-contained Master Notebook: [`Comparative_Prediction_Models_Master.ipynb`](Comparative%20Prediction%20Models/notebooks/Comparative_Prediction_Models_Master.ipynb).
- Incorporates EDA, feature scaling, model training, feature importance visualizations, comparative summary charts, and interactive sample inference (fully compatible with Google Colab & local Jupyter).

### 3. Classification Models Evaluated
- **Logistic Regression**
- **Support Vector Machine (SVM)**
- **Random Forest Classifier**
- **XGBoost Classifier**
- **K-Nearest Neighbors (KNN)** & **Decision Trees**

### 4. Documentation & Step-by-Step User Guides
- Updated both `README.md` files with step-by-step installation instructions, dependencies (`requirements.txt`), terminal execution commands, and notebook execution guides.

## How to Test
1. Checkout the branch & install dependencies:
   ```bash
   git checkout flood-prediction-system
   pip install -r requirements.txt



### Ongoing Improvements

Future improvements planned include:
*Increase in accuracy
* Hyperparameter tuning
* Advanced feature scaling
* Feature engineering
* Additional ensemble methods
* Improved dataset preprocessing

This project is also intended to serve as a beginner-friendly reference for learning and understanding major Machine Learning classification algorithms through a real-world environmental prediction use case.

###Related issue

Closes #1437

### CODE REVIEW

**Comment by v4rshh** [comment]
Hii @Niketkumardheeryan ,
I noticed that there is already a flood prediction model in the repo ,but here is the thing .
this is not a duplicate of that repo as this is made using a different dataset ,and this mainly helps beginners learn most of the classification ML models at a single stop.
 Hoping this gets noticed and gets helpful.

**Comment by Sibam-Paul** [comment]
@v4rshh make changes to the existing one don't add extra please 
If u can make it work with the old dataset it would be nice

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
Make changes to the existing one instead

**Comment by v4rshh** [comment]
> @v4rshh make changes to the existing one don't add extra please If u can make it work with the old dataset it would be nice

OK,THEN should i implement the same with the existing dataset??

**Comment by Sibam-Paul** [comment]
https://github.com/Niketkumardheeryan/ML-CaPsule/tree/master/Flood_Prediction

**Comment by Sibam-Paul** [comment]
@v4rshh come to vc of gssoc and pin me in the general I need to tell u what to do

**Comment by v4rshh** [comment]
> @v4rshh come to vc of gssoc and pin me in the general I need to tell u what to do

ur handle in discord??

**Comment by Sibam-Paul** [comment]
specter102 @v4rshh

**Comment by v4rshh** [comment]
@Sibam-Paul done the requested changes

**Review by Sibam-Paul — APPROVED** [review]
LGTM !
